### PR TITLE
Concurrent modification exception

### PR DESCRIPTION
**Pull request by djih**
* Following discussion on: https://github.com/candrews/HttpResponseCache/issues/22, best solution is to replace Android implementation of MessageDigest with alternate.
* Using MD5 implementation from: http://org.rodage.com/pub/java/security/MD5.java
* Instrumented following: https://github.com/candrews/HttpResponseCache/commit/eaeaeb2ae4f7905469d4916b8bbc9cf4506b4e30
* Added regression tests that validate new MD5 against Android MessageDigest

### CODE REVIEW

File: test/com/amplitude/security/MD5Test.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        compareMD5(preImage);
+    }
+
+    @Test
+    public void testMD5WithRandomStrings() {
+        for (int i = 0; i < 5; i++) {
+            compareMD5(UUID.randomUUID().toString());
+        }
+    }
+}
```
Comment (curtisliu) [inline test/com/amplitude/security/MD5Test.java]: Add a test for unicode and multi-byte unicode characters (similar to https://github.com/amplitude/Amplitude-Javascript/blob/master/test/md5.js)

**Comment by djih** [comment]
Added unicode tests, ptal

**Comment by curtisliu** [comment]
LGTM
